com_migratetojoomla/media/js/admin-migratetojoomla.js

```diff
@@ -0,0 +1,30 @@
+(function () {
```

Comment:
Don't overcomplicate the script
Start with copyright stuff, then add 
```
if (!Joomla) {
  throw new Error('Joomla API is not properly initialised');
}
```

No need for both englobing function() { 

Start your code with function addsteps() { ...